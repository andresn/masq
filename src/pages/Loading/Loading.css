.LoadingPage {
  background-color: var(--blue-500);
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.LoadingPage p {
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  text-align: center;
  color: white;
  opacity: 0.8;
}
