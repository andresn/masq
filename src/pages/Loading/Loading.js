import React from 'react'

import { LoadingDots, Avatar } from 'components'
import { Background, Logo } from 'icons'

import './Loading.css'

const styles = {
  background: {
    bottom: 0,
    maxHeight: 252,
    position: 'absolute'
  },
  logo: {
    paddingBottom: 64
  }
}

export default function Loading (props) {
  const { user } = props

  return (
    <div className='LoadingPage'>
      <Logo style={styles.logo} />
      {user &&
        <div className='user'>
          <Avatar user={user} image={user.image} />
          <p className='username'>{user.username}</p>
        </div>
      }
      <LoadingDots />
      <p>Loading</p>
      <Background style={styles.background} />
    </div>
  )
}
