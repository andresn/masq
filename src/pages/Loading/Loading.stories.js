import React from 'react'
import { storiesOf } from '@storybook/react'

import Loading from './Loading'

storiesOf('Loading Page', module)
  .add('without user', () => (
    <div style={{ height: 800 }}>
      <Loading />
    </div>
  ))
  .add('with a user', () => {
    const user = {
      image: 'https://randomuser.me/api/portraits/men/1.jpg',
      firstname: 'Jeffrey',
      lastname: 'Hoffman',
      username: 'Jeff'
    }

    return (
      <div style={{ height: 800 }}>
        <Loading user={user} />
      </div>
    )
  })
