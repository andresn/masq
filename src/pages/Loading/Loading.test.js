import React from 'react'
import renderer from 'react-test-renderer'
import Loading from './Loading'

import { UserContext } from 'context/user'

it('renders Loading page', () => {
  const user = {
    image: 'https://randomuser.me/api/portraits/women/79.jpg',
    firstname: 'Alicia',
    lastname: 'Ford',
    username: 'Mustang'
  }

  const component = renderer.create(
    <UserContext.Provider value={user}>
      <Loading user={user} />
    </UserContext.Provider>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
