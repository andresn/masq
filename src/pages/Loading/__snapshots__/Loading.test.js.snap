// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Loading page 1`] = `
<div
  className="LoadingPage"
>
  <div
    className="content"
  >
    <div
      className="Avatar"
      style={
        Object {
          "backgroundImage": "url(https://randomuser.me/api/portraits/women/79.jpg)",
        }
      }
    />
    <h1>
      Mustang
    </h1>
    <div
      id="separator"
      style={
        Object {
          "marginTop": "16px",
        }
      }
    />
    <div
      className="Loading"
    >
      <div
        className="loading-dots"
        style={
          Object {
            "backgroundColor": "#4ebbc2",
          }
        }
      />
      <div
        className="loading-dots"
        style={
          Object {
            "backgroundColor": "#3394d2",
          }
        }
      />
      <div
        className="loading-dots"
        style={
          Object {
            "backgroundColor": "#e53b5b",
          }
        }
      />
    </div>
    <p>
      Connecting ...
    </p>
  </div>
</div>
`;
